vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java

```diff
@@ -238,7 +238,8 @@ private void setupDataProviderListener(DataProvider<T, ?> dataProvider) {
     @Override
     public void setValue(T value) {
         super.setValue(value);
-        getRadioButtons().forEach(rb -> rb.setChecked(rb.getItem().equals(value)));
+        getRadioButtons().forEach(rb -> rb.setChecked(
+            rb.getItem() == value || (rb.getItem() != null && rb.getItem().equals(value))));
```

Comment:
Could this be changed to:
```suggestion
        getRadioButtons().forEach(rb -> rb.setChecked(Object.equals(rb.getItem(), value)));
```

Reply:
Yup, nice